### PR TITLE
#184347038 view cart

### PR DESCRIPTION
**Pull request by irakozetony**
### What does this PR do?
Users should be able to view their cart

### Description of Task to be completed?
Return items in the cart and their total to the user

### How should this be manually tested?
1. Checkout to feature/view-cart-184347038
2. Run `npm run seeding`
3. Run `npm run dev`
4. Add products to your cart at `http://localhost:<port>/api/v1/cart/add-to-cart` URL in Postman
5. View products in the cart at `http://localhost:<port>/api/v1/cart/view-cart` URL in Postman

#### What are the relevant pivotal tracker stories?
[#184347038](https://www.pivotaltracker.com/story/show/184347038)

### CODE REVIEW

File: src/controllers/cart.controller.js [hunk 1]
```diff
@@ -26,4 +27,28 @@ export class CartController {
 			});
 		}
 	}
+
+	static async viewCart(req, res) {
+		try {
+			const cart = req.cart;
+			const cartProducts = await CartService.getCartProducts(cart.products);
```
Comment (ncutixavier) [inline src/controllers/cart.controller.js]: Service to get products by ids (`getCartProducts`) should only accept an array of ids, not a cart array

File: src/database/config/config.js [hunk 2]
```diff
@@ -1,26 +1,25 @@
...
-     logging:false,
-  },
-  production: {
-    url: process.env.DATABASE_URL,
-    dialect: 'postgres',
-  },
+	development: {
+		url: process.env.DEV_DATABASE_URL,
+		dialect: 'postgres',
+		//   logging:false
```
Comment (ncutixavier) [inline src/database/config/config.js]: Let's get rid of unnecessary comments

File: src/routes/api/cart.routes.js [hunk 3]
```diff
@@ -17,4 +17,12 @@ cartRoutes.post(
...
 );
+
+cartRoutes.get(
+  '/view-cart',
+  protectRoute,
+  restrictTo('buyer', 'admin'),
+  getUserCart,
+  CartController.viewCart
+);
 export default cartRoutes;
```
Comment (ncutixavier) [inline src/routes/api/cart.routes.js]: I want you to refactor endpoints for cart to be `api/v1/carts`

**Review by ncutixavier — CHANGES_REQUESTED** [review]

**Review by fabmukunzi — APPROVED** [review]

File: src/documentation/cart.docs.js [hunk 4]
```diff
@@ -1,46 +1,55 @@
...
-  },
-  consumes: ['application/json'],
-  responses,
-  security: [
-    {
-      bearerAuth: [],
-    },
-  ],
+const carts = {
+	'/api/v1/chats': {
```
Comment (ncutixavier) [inline src/documentation/cart.docs.js]: Why do you have chats in cart?

File: src/controllers/cart.controller.js [hunk 5]
```diff
@@ -26,4 +27,29 @@ export class CartController {
 			});
 		}
 	}
+
+	static async viewCart(req, res) {
+		try {
+			const cart = req.cart;
+			const { products } = cart;
+			const cartProducts = await CartService.getCartProducts(products);
```
Comment (ncutixavier) [inline src/controllers/cart.controller.js]: I didn't see any changes in this service. Just pass on ids, not cart

**Review by ncutixavier — CHANGES_REQUESTED** [review]

**Review by ncutixavier — APPROVED** [review]
LGTM

**Pull request merged**